contracts/contracts/TACoEvidence.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// Assuming the use of OpenZeppelin contracts for secure cryptographic operations
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "../threshold/ITACoChildApplication.sol";
+
+contract TACoEvidence {
+    using ECDSA for bytes32;
+
+    // Event declarations for logging
+    event CollectionStarted(
+        uint32 indexed id,
+        uint32 indexed startTime,
+        uint32 endTime,
+        uint256 nonce
+    );
+
+    struct Submission {
+        address operator;
+        bytes32 evidence;
+    }
+    struct Collection {
+        uint32 initTimestamp;
+        uint32 endTimestamp;
+        uint256 nonce;
+        mapping(address => Submission[]) submissions;
+    }
+
+    ITACoChildApplication public immutable application;
+
+    uint256 public submissionWindow = 1 hours;
+    Collection[] public collections;
+
+    constructor() {
+        admin = msg.sender;
```

Comment:
This isn't defined, right? Anyway, I'd define as immutable too.

Reply:
I'd change it to Ownable and owner